lib/api/v1/users.rb

```diff
@@ -0,0 +1,204 @@
+  module V1
+    class Users < Grape::API
+      version "v1", using: :path
+
+      resource :users do
+        before do
+          authorization!
+        end
+
+        route_param :id, type: Integer do
+          resource :application_tokens do
+            # List application tokens beloged to user with given :id.
+            desc "Returns list of user's tokens.",
+              params:   API::Entities::Users.documentation.slice(:id),
+              is_array: true,
+              entity:   API::Entities::ApplicationTokens,
+              failure:  [
+                [401, "Authentication fails."],
+                [403, "Authorization fails."],
+                [404, "Not found."]
+              ]
+
+            get do
+              user = User.find params[:id]
+              present user.application_tokens,
+                with: API::Entities::ApplicationTokens
+            end
+
+            # Create application token for user with given :id.
+            desc "Create user's token.",
+              params:   API::Entities::Users.documentation.slice(:id),
+              success:  { code: 200 },
+              entity:   API::Entities::ApplicationTokens,
+              failure:  [
+                [400, "Bad request.", API::Entities::ApiErrors],
+                [401, "Authentication fails."],
+                [403, "Authorization fails."]
+              ],
+              consumes: ["application/x-www-form-urlencoded"]
+
+            params do
+              requires :application, documentation: { desc: "Application name" }
+            end
+
+            post do
+              application_token, plain_token = ApplicationToken.create_token(
+                current_user: @user,
+                user_id:      params[:id],
+                params:       declared(params.slice(:application))
+              )
+
+              if application_token.errors.empty?
+                status 200
+                { plain_token: plain_token }
+              else
+                status 400
+                { errors: application_token.errors }
+              end
+            end
+          end
+        end
+
+        resource :application_tokens do
+          desc "Delete application token.",
+            failure: [
+              [401, "Authentication fails."],
+              [403, "Authorization fails."],
+              [404, "Not found."]
+            ]
+
+          params do
+            requires :id, documentation: { desc: "Token id" }
+          end
+
+          delete ":id" do
+            token = ApplicationToken.find(params[:id])
+            token.create_activity! :destroy, @user
+            token.destroy
+            status 204
+          end
+        end
+
+        desc "Create new user.",
+          failure:  [
+            [400, "Bad request.", API::Entities::ApiErrors],
+            [401, "Authentication fails."],
+            [403, "Authorization fails."]
+          ],
+          consumes: ["application/x-www-form-urlencoded"]
```

Comment:
Maybe this can be done in another PR, but would it be too difficult to also support JSON ? It would be simpler than with an encoded form. See the curl command I had to use:

```
curl -X POST \
     -H "PORTUS-AUTH: mssola:KXsgWZQyDq8rMoy-hHaD" \
     -F "user[username]=another" -F "user[email]=asd@example.org" -F "user[password]=12341234" \
     https://registry.mssola.cat/api/v1/users
```

Maybe this could work too (so without the `user[...]` part):

```
curl -X POST \
     -H "PORTUS-AUTH: mssola:KXsgWZQyDq8rMoy-hHaD" \
     https://registry.mssola.cat/api/v1/users?username=another&email=asd@example.org&password=12341234
```